### DISCUSSION TITLE
v3.3.0: Add DAQP solver, Solution.found attribute

### DISCUSSION THREAD

**Discussion by stephane-caron** [qpsolvers/qpsolvers#203] [Announcements]
This release adds [DAQP](https://darnstrom.github.io/daqp/), a dual active-set solver suited to small/medium-scale dense problems. It also introduces the ``Solution.found`` boolean, which is now used rather than ``Solution.x is not None`` to check if a call to ``solve_problem`` was successful. Solvers can then return a non-empty ``Solution.x`` even when the optimal solution was not found (*e.g.* an intermediate value reached after a maximum number of iterations).

Thanks to @darnstrom for implementing the DAQP interface, and @rxian for suggesting the switch to ``Solution.found`` :+1: 

### Added

- New sample problems in ``qpsolvers.problems``
- New solver: [DAQP](https://darnstrom.github.io/daqp/)

### Changed

- Dual multipliers are empty arrays rather than None when no constraint
- Store solver results even when solution is not found
- Use ``Solution.found`` as success signal rather than ``Solution.x is None``

### Fixed

- Unit test on actual solution to QPSUT03 problem

<hr /><em>This discussion was created from the release <a href='https://github.com/qpsolvers/qpsolvers/releases/tag/v3.3.0'>v3.3.0</a>.</em>

**Comment by microprediction** [comment]
I noticed this broke my install on mac (m2). When I have a chance I will try to get it qadp to install and provide some more helpful feedback.

**Reply by stephane-caron** [reply]
Okay, this could be related to the addition of DAQP to dependencies (@darnstrom for your information). Thank you for reporting this :+1:

**Reply by darnstrom** [reply]
@microprediction @stephane-caron I found that a Cython file was not uploaded to PyPI in the 0.5.0 release of daqp (due to a typo in MANIFEST.in). Since this will yield problems if a wheel is not available, this could be the reason. The missing file is added in v0.5.1 of daqp, so ensuring that this version is installed with pip might resolve the problem.
